File: pueblo_golf_tournament_mobile/lib/dto/request/setup-tournament-schedules-request-dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'setup-tournament-schedules-request-dto.g.dart';

@JsonSerializable()
class SetupTournamentSchedulesRequestDto {
  final int tournamentId;
  final List<ScheduleDto> schedules;

  SetupTournamentSchedulesRequestDto(
      {required this.tournamentId, required this.schedules});

  factory SetupTournamentSchedulesRequestDto.fromJson(
          Map<String, dynamic> json) =>
      _$SetupTournamentSchedulesRequestDtoFromJson(json);
  Map<String, dynamic> toJson() =>
      _$SetupTournamentSchedulesRequestDtoToJson(this);
}

@JsonSerializable()
class ScheduleDto {
  final DateTime date;
  final List<TimeSlotDto> timeSeries;

  ScheduleDto({required this.date, required this.timeSeries});

  factory ScheduleDto.fromJson(Map<String, dynamic> json) =>
      _$ScheduleDtoFromJson(json);
  Map<String, dynamic> toJson() => _$ScheduleDtoToJson(this);
}

@JsonSerializable()
class TimeSlotDto {
  final DateTime timeSeriesSlot;
  final bool isEnabled;
  final bool isSelected;

  TimeSlotDto(
      {required this.timeSeriesSlot,
      required this.isEnabled,
      required this.isSelected});

  factory TimeSlotDto.fromJson(Map<String, dynamic> json) =>
      _$TimeSlotDtoFromJson(json);
  Map<String, dynamic> toJson() => _$TimeSlotDtoToJson(this);
}

File: pueblo-golf-tournament-api/Dto/Incoming/SetupTournamentSchedulesRequestDto.cs
namespace pueblo_golf_tournament_api.Dto.Incoming
{
    public class SetupTournamentSchedulesRequestDto
    {
        public long TournamentId { get; set; }
        public List<Schedule> Schedules { get; set; } = new List<Schedule>();
    }

    public class Schedule
    {
        public DateTime Date { get; set; }
        public List<TimeSlot> TimeSeries { get; set; } = new List<TimeSlot>();
    }

    public class TimeSlot
    {
        public DateTime TimeSeriesSlot { get; set; }
        public bool IsEnabled { get; set; }
        public bool IsSelected { get; set; }
    }
}